src/modules/logger/logger.cpp

```diff
@@ -0,0 +1,926 @@
+		return;
+	}
+
+	if (!(_parameter_update_sub = new uORB::Subscription<parameter_update_s>(ORB_ID(parameter_update)))) {
+		delete _vehicle_status_sub;
+		_vehicle_status_sub = nullptr;
+		PX4_ERR("Failed to allocate subscription");
+		return;
+	}
+
+
+	add_topic("sensor_gyro", 0);
+	add_topic("sensor_accel", 0);
+	add_topic("vehicle_rates_setpoint", 10);
+	add_topic("vehicle_attitude_setpoint", 10);
+	add_topic("vehicle_attitude", 0);
+	add_topic("actuator_outputs", 50);
+	add_topic("battery_status", 100);
+	add_topic("vehicle_command", 100);
+	add_topic("actuator_controls", 10);
+	add_topic("vehicle_local_position_setpoint", 200);
+	add_topic("rc_channels", 20);
+//	add_topic("ekf2_innovations", 20);
+	add_topic("commander_state", 100);
+	add_topic("vehicle_local_position", 200);
+	add_topic("vehicle_global_position", 200);
+	add_topic("system_power", 100);
+	add_topic("servorail_status", 200);
+	add_topic("mc_att_ctrl_status", 50);
+//	add_topic("control_state");
+//	add_topic("estimator_status");
+	add_topic("vehicle_status", 200);
+
+	if (!_writer.init()) {
+		PX4_ERR("logger: init of writer failed");
+		return;
+	}
+
+	int ret = _writer.thread_start(_writer_thread);
+
+	if (ret) {
+		PX4_ERR("logger: failed to create writer thread (%i)", ret);
+		return;
+	}
+
+	_task_should_exit = false;
+
+#ifdef DBGPRINT
+	hrt_abstime	timer_start = 0;
+	uint32_t	total_bytes = 0;
+#endif /* DBGPRINT */
+
+	// we start logging immediately
+	// the case where we wait with logging until vehicle is armed is handled below
+	if (_log_on_start) {
+		start_log();
+	}
+
+	/* every log_interval usec, check for orb updates */
+	while (!_task_should_exit) {
+
+		// Start/stop logging when system arm/disarm
+		if (_vehicle_status_sub->check_updated()) {
+			_vehicle_status_sub->update();
+			bool armed = (_vehicle_status_sub->get().arming_state == vehicle_status_s::ARMING_STATE_ARMED) ||
+				     (_vehicle_status_sub->get().arming_state == vehicle_status_s::ARMING_STATE_ARMED_ERROR);
+
+			if (_enabled != armed && !_log_on_start) {
+				if (armed) {
+					start_log();
+
+#ifdef DBGPRINT
+					timer_start = hrt_absolute_time();
+					total_bytes = 0;
+#endif /* DBGPRINT */
+
+				} else {
+					stop_log();
+				}
+			}
+		}
+
+		if (_enabled) {
+
+			bool data_written = false;
+
+			/* Check if parameters have changed */
+			// this needs to change to a timestamped record to record a history of parameter changes
+			if (_parameter_update_sub->check_updated()) {
+				warnx("parameter update");
+				_parameter_update_sub->update();
+				write_changed_parameters();
+			}
+
+			// Write data messages for normal subscriptions
+			int msg_id = 0;
+
+			/* wait for lock on log buffer */
+			_writer.lock();
+
+			for (LoggerSubscription &sub : _subscriptions) {
+				/* each message consists of a header followed by an orb data object
+				 */
+				size_t msg_size = sizeof(message_data_header_s) + sub.metadata->o_size;
+				//TODO: use sub.metadata->o_size_no_padding
```

Comment:
So pad bytes are now copied into the log buffer and written to the logfile?
How is the Java code supposed to know where the padding is?